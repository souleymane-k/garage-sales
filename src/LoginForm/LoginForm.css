*{
  box-sizing: border-box;
  background-image: #ababab;
  /* background-image: url(../../souleymane-3.jpg);  */
  /* z-index: 1;  */
  

}
    label{
      margin: 1em 0 0 0;
      padding: .25em 0;
      text-align: left;
      width: auto;
    }

  
     input {
      width: 60%;;
      line-height: 2em;
      padding: 0 .5em;
      border-radius: 2px;
      border: 2px solid; 
    } 
  
    input:focus {
      border: 2px solid;
    }
  
    
    h3{
        text-align: center;
        color:black;
        margin:2px;
    }
   
   /* input[type='submit'] { */
    /* margin: 2em auto 1em auto; */
     /* margin-top:5px;
    padding: 0.25em 0;
    width: 60%; */
    /* background-color: black; */
    /* color:white;
    border: 1px solid whitesmoke;
    border-radius: 2px;
    box-shadow: 2px 2px 2px 0px black;
    cursor: pointer; 
    
   }  */
   .submitButton{
    /* position: absolute; */
    width: 100%;
    margin:5px auto;
    padding: 0.25em 0;
    /* width: 400px; */
    /* left: 600px; */
    height:40px;
    background-color: black;
    color:white;
    border: 1px solid whitesmoke;
    border-radius: 2px;
    /* box-shadow: 2px 2px 2px 0px black; */
    cursor: pointer;
   }
    
  label {
    margin: 1em 0 0 0;
    padding: .25em 0;
    text-align: left;
    width: auto;
    color:white;
  /*   display: flex;
    flex-direction: column nowrap; */
  }
  
.login-page {
  padding: 2em 0;
  margin: 0;
}


form {
  margin: 0 auto;
  /* // border: 1px solid black; */
}
.login-form{
  /* background-image: url(../../souleymane-3.jpg); */
  height:1000px;
  margin-top:0;
}

.loginForm{
  max-width: 60%;
  display: flex;
  height: 300px;
  margin:30 auto;
  flex-flow: column nowrap;
  background-color: #EAF6F6;
  /* // justify-content: center;
  // align-items: center; */
  background-color: #E6E6FA;
  border: none;
  border-radius: 4px;
  box-shadow: -1px 1px 5px 4px #000000;
  padding: 1em 1.5em;
}

  label {
    margin: 1em 0 0 0;
    padding: .25em 0;
    text-align: left;
    width: auto;
    color:black
  /*   display: flex;
    flex-direction: column nowrap; */
  }

  input {
    width: 100%;
    line-height: 2em;
    padding: 0 .5em;
    border-radius: 2px;
    border: 2px solid ;
    /* background-color: ; */
  }

  input:focus {
    border: 2px solid red;
  }

  /* div {
    margin: 1em;
  } */


.error-msg {
  color: color-error;
}

.warning-msg {
  margin-top: 1em;
  color: color-primary;
  font-size: 12px;
}

input[type='submit'] {
  margin: 2em auto 1em auto;
  padding: 0.25em 0;
  width: 100px;
  background-color: color-overlay;
  border: 1px solid color-surface-light;
  border-radius: 2px;
  box-shadow: 2px 2px 2px 0px color-shadow;
  transition: box-shadow 0.1s;
}
input:hover {
    background-color: red;
    color: white;
    border: 1px solid;
    cursor: pointer;
  }
  input:focus {
    background-color: red;
    color: color-text-normal;
    border: 1px solid color-quaternary;
    box-shadow: none;
  }
  input:active {
    background-color: color-overlay-dark;
    color: color-text-normal;
    border: 1px solid color-secondary;
  }



/* ............@media Query...........................*/

@media all and (max-width: 640px) {
  .submitButton{
    /* position: relative; */
    margin:0 auto;
    padding: 0.25em 0;
    width:100%;
    /* left:150px; */
    height:40px;
    background-color: black;
    color:white;
    /* border: 1px solid whitesmoke; */
    border-radius: 2px;
    box-shadow: 2px 2px 2px 0px black;
    cursor: pointer;
   }
   .loginform{
    position: absolute;
    width: 80%;
    margin: 40 auto;
    color:#EAF6F6;
    /* display: flex; */
    height: 200px;
    flex-flow: column nowrap;
    /* // justify-content: center;
    // align-items: center; */
    background-color: #E6E6FA;
    border: none;
    border-radius: 4px;
    box-shadow: -1px 1px 5px 4px #000000;
    padding: 1em 1.5em;
  }
   

 } 

