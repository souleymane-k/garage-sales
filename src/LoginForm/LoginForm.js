import React, { Component } from 'react'
import AuthApiService from '../services/auth-api-service'
import ApiContext from '../contexts/ApiContext';
import TokenService from '../services/TokenService'
import './LoginForm.css'


export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }
  static contextType = ApiContext;
  state = { error: null }

  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    const {username, password} = ev.target
    console.log('login form submitted')
     console.log({ username, password })

    AuthApiService.postLogin(
      username.value,
      password.value,
    )
      .then(res => {
  
        TokenService.saveAuthToken(res.authToken)
        username.value = ''
        this.context.setUser(true)
        password.value = ''
        console.log('hello world')
        this.props.onLoginSuccess()
      })
      .catch(res => {
        console.log(res)
        this.setState({ error: res.error })
      })
  }

  render() {
    // console.log('login form')
    const { error } = this.state
    // console.log(error)
    return (
      <form className='login-form' onSubmit={this.handleSubmitJwtAuth}>
        <div role='alert'>
          {error && <p className='red'>{error.message}</p>}
        </div>
        <fieldset className="loginForm">
        <div className='form-group'>
          <label htmlFor='LoginForm__username'>
            Username
          </label>
          <input
            required
            name='username'
            type="text"
            placeholder='username'
            id='LoginForm__username'>
          </input>
        </div>
        <div className='form-group'>
          <label htmlFor='LoginForm__password'>
            Password
          </label>
          <input
            required
            name='password'
            type='password'
            placeholder='password'
            id='LoginForm__password'>
          </input>
        </div>
        <div className='form-controls'>
        <button   className='submitButton' type='submit'>Login</button>
       </div>
        </fieldset>
      </form>
    )
  }
}











//////
// import React, { Component } from 'react'
// import { Button, Input } from '../Direction/Direction'

// export default class LoginForm extends Component {
//   static defaultProps = {
//     onLoginSuccess: () => {}
//   }

//   state = { error: null }

//   handleSubmitBasicAuth = ev => {
//     ev.preventDefault()
//     const { username, password } = ev.target

//     console.log('login form submitted')
//     console.log({ username, password })

//     username.value = ''
//     password.value = ''
//     this.props.onLoginSuccess()
//   }

//   render() {
//     const { error } = this.state
//     return (
//       <form
//         className='LoginForm'
//         onSubmit={this.handleSubmitBasicAuth}
//       >
//         <div role='alert'>
//           {error && <p className='red'>{error}</p>}
//         </div>
//         <div className='username'>
//           <label htmlFor='LoginForm__username'>
//             Username
//           </label>
//           <Input
//             name='username'
//             id='LoginForm__username'>
//           </Input>
//         </div>
//         <div className='password'>
//           <label htmlFor='LoginForm__password'>
//             Password
//           </label>
//           <Input
//             name='password'
//             type='password'
//             id='LoginForm__password'>
//           </Input>
//         </div>
//         <Button type='submit'>
//           Login
//         </Button>
//       </form>
//     )
//   }
// }





