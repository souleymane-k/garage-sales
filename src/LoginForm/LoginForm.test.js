import React from 'react';
import { shallow } from 'enzyme'
import ReactDOM from 'react-dom'
import toJson from 'enzyme-to-json'
import LoginForm from './LoginForm';

describe(`LoginForm component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})

