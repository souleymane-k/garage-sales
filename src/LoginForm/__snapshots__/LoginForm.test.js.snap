// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the complete form 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <fieldset
    className="loginForm"
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="LoginForm__username"
      >
        Username
      </label>
      <input
        id="LoginForm__username"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <input
        id="LoginForm__password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-controls"
    >
      <button
        className="submitButton"
        type="submit"
      >
        Login
      </button>
    </div>
  </fieldset>
</form>
`;
